cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsLaunchTemplateImageUpdateService.java

```diff
@@ -1,58 +1,66 @@
-        try {
-            cloudFormationClient.updateStack(updateStackRequest);
-        } catch (AmazonCloudFormationException e) {
-            if ("ValidationError".equals(e.getErrorCode()) && e.getErrorMessage().contains("No updates are to be performed")) {
-                LOGGER.debug(String.format("CloudFormation is not updated as it is already in the latest state, name: %s", cfResource.getName()), e);
-            } else {
-                throw e;
-            }
+        AmazonAutoScalingClient autoScalingClient = awsClient.createAutoScalingClient(credentialView, regionName);
+        AmazonEc2Client ec2Client = awsClient.createEc2Client(credentialView, regionName);
+        Map<AutoScalingGroup, String> autoScalingGroups = autoScalingGroupHandler.getAutoScalingGroups(cloudFormationClient, autoScalingClient, cfResource);
+        for (Map.Entry<AutoScalingGroup, String> asgEntry : autoScalingGroups.entrySet()) {
+            AutoScalingGroup autoScalingGroup = asgEntry.getKey();
+            LaunchTemplateSpecification launchTemplateSpecification = autoScalingGroup.getLaunchTemplate() == null ?
+                    autoScalingGroup.getMixedInstancesPolicy().getLaunchTemplate().getLaunchTemplateSpecification() : autoScalingGroup.getLaunchTemplate();
+            CreateLaunchTemplateVersionRequest createLaunchTemplateVersionRequest = new CreateLaunchTemplateVersionRequest()
+                    .withLaunchTemplateId(launchTemplateSpecification.getLaunchTemplateId())
+                    .withSourceVersion(launchTemplateSpecification.getVersion())
+                    .withLaunchTemplateData(new RequestLaunchTemplateData().withImageId(stack.getImage().getImageName()));
+            CreateLaunchTemplateVersionResult createLaunchTemplateVersionResult = ec2Client.createLaunchTemplateVersion(createLaunchTemplateVersionRequest);
+            ModifyLaunchTemplateRequest modifyLaunchTemplateRequest = new ModifyLaunchTemplateRequest()
+                    .withLaunchTemplateId(launchTemplateSpecification.getLaunchTemplateId())
+                    .withDefaultVersion(createLaunchTemplateVersionResult.getLaunchTemplateVersion().getVersionNumber().toString());
+            ec2Client.modifyLaunchTemplate(modifyLaunchTemplateRequest);
+            LaunchTemplateSpecification newLaunchTemplateSpecification = new LaunchTemplateSpecification()
+                    .withLaunchTemplateId(launchTemplateSpecification.getLaunchTemplateId())
+                    .withVersion(createLaunchTemplateVersionResult.getLaunchTemplateVersion().getVersionNumber().toString());
+            autoScalingClient.updateAutoScalingGroup(new UpdateAutoScalingGroupRequest()
+                    .withAutoScalingGroupName(autoScalingGroup.getAutoScalingGroupName())
+                    .withLaunchTemplate(newLaunchTemplateSpecification));
```

Comment:
could you split these into methods and add logging and result validations, like:
- `createNewLaunchTemplateVersion`
- `configureASGForNewVersion`

I would expect the `for` body has 3-4 statements at max, or even that one could be moved to a single method so the `updateImage` won't have too many lines.